File: Programming Basics/C++/While loops/Cake/Cake.cpp
#include <iostream>
#include <string>
using namespace std;

int main()
{
	int width, length;
	cin >> width >> length;

	int totalPieces = width * length;
	bool cakeIsEnough = true;

	string input;
	while (cin >> input && input != "STOP") {
		int piecesToTake = stoi(input);
		totalPieces -= piecesToTake;

		if (totalPieces < 0) {
			cakeIsEnough = false;
			break;
		}
	}

	if (cakeIsEnough) cout << totalPieces << " pieces are left." << endl;
	else cout << "No more cake left! You need " << abs(totalPieces) << " pieces more." << endl;
}

File: Programming Basics/JavaScript/While loops/Labs/Exercises/cake.js
function solve(input) {
    const width = Number(input[0]);
    const length = Number(input[1]);

    let totalPieces = width * length;
    let cakeIsEnough = true;
    let index = 2;

    while (input[index] != "STOP") {
        const piecesToTake = Number(input[index]);
        totalPieces -= piecesToTake;

        if (totalPieces < 0) {
            cakeIsEnough = false;
            break;
        }

        index++;
    }

    if (cakeIsEnough) console.log(`${totalPieces} pieces are left.`);
    else console.log(`No more cake left! You need ${Math.abs(totalPieces)} pieces more.`);
}
